autotest/pyscripts/test_gdal_retile.py

```diff
@@ -241,7 +240,64 @@ def test_gdal_retile_4():
         assert ds.RasterYSize == height, filename
         ds = None
 
-    
+
+###############################################################################
+# Test gdal_retile.py with input having a NoData value
+
+
+def test_gdal_retile_5():
+
+    try:
+        from osgeo import gdalnumeric
+        gdalnumeric.zeros
+        import numpy as np
+    except (ImportError, AttributeError):
+        pytest.skip()
+
+    nodata_value = -3.4028234663852886e+38
+    raster_array = np.array(([0.0, 2.0], [-1.0, nodata_value]))
+
+    script_path = test_py_scripts.get_py_script('gdal_retile')
+    if script_path is None:
+        pytest.skip()
+
+    drv = gdal.GetDriverByName('GTiff')
+    srs = osr.SpatialReference()
+    srs.SetWellKnownGeogCS('WGS84')
+    wkt = srs.ExportToWkt()
+
+    ds = drv.Create('tmp/in5.tif', 2, 2, 1, gdal.GDT_Float32)
+    px1_x = 0.1 / ds.RasterXSize
+    px1_y = 0.1 / ds.RasterYSize
+    ds.SetProjection(wkt)
+    ds.SetGeoTransform([0, px1_x, 0, 30, 0, -px1_y])
+    raster_band = ds.GetRasterBand(1)
+    raster_band.SetNoDataValue(nodata_value)
+    raster_band.WriteArray(raster_array)
+    raster_band = None
+    ds = None
+
+    try:
+        os.mkdir('tmp/outretile5')
+    except OSError:
+        pass
+
+    test_py_scripts.run_py_script(script_path, 'gdal_retile', '-v -targetDir tmp/outretile5 tmp/in5.tif')
+
+    ds = gdal.Open('tmp/outretile5/in5_1_1.tif')
+    raster_band = ds.GetRasterBand(1)
+
+    assert raster_band.GetNoDataValue() == nodata_value, \
+        ('Wrong nodata value.\nExpected %f, Got: %f' % (nodata_value, raster_band.GetNoDataValue()))
+
+    min_val, max_val = raster_band.ComputeRasterMinMax()
+    assert max_val, \
+        ('Wrong maximum value.\nExpected 2.0, Got: %f' % max_val)
+
+    assert min_val == -1.0, \
+        ('Wrong minimum value.\nExpected -1.0, Got: %f' % min_val)
+
```

Comment:
You may need to add an explicit ds = None to force dataset closing to fix the `WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'tmp/outretile5\\in5_1_1.tif'` of 
https://ci.appveyor.com/project/OSGeo/gdal/builds/22624606/job/6jfgqtwy18yvw4nu

Reply:
Thanks for keeping on top of this. Is there a way to receive some sort of notification when the builds fail, so I don't forget about the pull request until your next comment pops up in my inbox?

Reply:
I'm not aware of one. You have to monitor regularly the progress of the builds